parlai/tasks/math_dataset/agents.py

```diff
@@ -109,7 +109,10 @@ def get_data_for_fold(self, fold):
                 rand_steps = self._clean_steps(
                     self.math_random.choice(data)["solution"]
                 ).split(". ")
-                random_step = self.math_random.choice(rand_steps)
+                random_step = ""
+                # find non-empty random step
+                while random_step == "" or random_step == " ":
```

Comment:
Any chance this could infinite loop (cause all steps are "", in case someone tries to mix perturbations)?